docs/src/tutorials/write-my-software-management-plugin.md

```diff
@@ -0,0 +1,393 @@
+# Write my software management plugin
+
+**thin-edge.io** provides Software Management plugins natively for APT (Debian).
+However, there are many package management systems in the world,
+and you may want to have a plugin that is suitable for your device.
+For such a demand, we provide **Software Management Plugin API**
+to write a custom Software Management plugin in your preferred language.
+
+In this tutorial, we will look into the **Software Management Plugin API**,
+and learn how to write your own plugin with a docker plugin shell script example.
+
+## Create a plugin
+
+Create a _docker_ file in the directory _/etc/tedge/sm-plugins/_. 
+A plugin must be an executable file and located in the directory.
+
+Filename: /etc/tedge/sm-plugins/docker
+
+```shell
+#!/bin/bash
+
+set -e
```

Comment:
This will make the script to return exit code of failed command and this is not what the API spec advices.

You want to use `|| exit 2` on each command to return correct code.

Reply:
Very good point :)